src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKStyle.java

```diff
@@ -182,6 +182,14 @@ else if (type == GTKColorType.WHITE) {
             }
         }
 
+        if (id == Region.LABEL && type == ColorType.FOREGROUND
+                && (state & SynthConstants.ENABLED) != 0) {
+            Color c = (Color) get(context, "Label.foreground");
+            if (c != null) {
```

Comment:
Why is it different for Label..We can have foreground color for RadioButton, Checkboxes similar to [JDK-8298083](https://bugs.openjdk.org/browse/JDK-8298083) and there it was not needed..

Reply:
I will check and revert back.

Reply:
test bug4314194 mentioned in [JDK-8298083](https://bugs.openjdk.org/browse/JDK-8298083) fails in GTK LAF and rightly it is problemlisted for linux machine. @mrserb has created a bug [JDK-8298153](https://bugs.openjdk.org/browse/JDK-8298153).

So, I think we need to check if color is set by user explicitly in GTK LAF.

Reply:
So, please check if the fix can be extended for checkbox and radiobutton

Reply:
The similar approach is working for Checkbox and Radiobutton. Should I raise a separate PR or extend the fix here?

Reply:
you can ask @mrserb if you can assign [JDK-8298153](https://bugs.openjdk.org/browse/JDK-8298153). to you and then add that issue to this and add those widgets fix also here, in my opinion..

Reply:
> @mrserb [JDK-8298153](https://bugs.openjdk.org/browse/JDK-8298153) issue raised by you for JCheckBox and JRadioButton has similar issue with JLabel. The proposed fix here resolves the issue for JLabel and similar approach is applicable for JCheckBox and JradioButton as well.
> 
> If you are ok then I will assign [JDK-8298153](https://bugs.openjdk.org/browse/JDK-8298153) to me and propose a fix.

@mrserb What do you think?